src/avlizer.erl

```diff
@@ -25,17 +25,59 @@
         ]).
 
 start(_StartType, _StartArgs) ->
-  case osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false) of
-    false -> ok;
-    URL ->
-      Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
-      Vars = Vars0#{schema_registry_url => URL},
-      application:set_env(?APPLICATION, avlizer_confluent, Vars)
-  end,
+  set_envs(),
   avlizer_sup:start_link().
 
 stop(_State) -> ok.
 
+set_envs() ->
+  Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
+  Vars1 = maybe_put_schema_registry_url(Vars0),
+  Vars2 = maybe_put_schema_registry_auth(Vars1),
+  application:set_env(?APPLICATION, avlizer_confluent, Vars2).
+
+maybe_put_schema_registry_url(Map) ->
+  SchemaRegistryURL = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false),
+  case SchemaRegistryURL of
+    false -> Map;
+    URL -> maps:put(schema_registry_url, URL, Map)
+  end.
+
+maybe_put_schema_registry_auth(Map) ->
+  SchemaRegistryAuthMechanism = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_MECHANISM", false),
+  SchemaRegistryAuthUsername = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_USERNAME", false),
+  SchemaRegistryAuthPassword = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_PASSWORD", false),
+
+  Auth = {maybe_string_to_atom(SchemaRegistryAuthMechanism), SchemaRegistryAuthUsername, SchemaRegistryAuthPassword},
+
+  Map0 = case lists:member(false, tuple_to_list(Auth)) of
+    true -> Map;
```

Comment:
we should perhaps perform a more strict check:  either all or none are set.
i.e. crash or at least log an error if any one of them is missing.

Reply:
Good idea, I've created a new function to handle this scenario: `osenv_all/2`